src/middlewares/checks.ts

```diff
@@ -0,0 +1,222 @@
+import { User, Organisation, Booking, Venue } from '@prisma/client'
+import { BookingPayload } from '../services/bookings'
+import {
+  DURATION_PER_SLOT,
+  MAX_SLOTS_PER_BOOKING,
+  MIN_SLOTS_BETWEEN_BOOKINGS,
+  MIN_SLOTS_PER_BOOKING,
+} from '../configs/common'
+import prisma from '../services/db'
+
+/**
+ * Checks if user is in the organisation OR if the user is an Admin
+ *
+ * @param user
+ * @param organisation
+ * @returns true if user is in the organisation
+ */
+export async function checkUserinOrg(
+  user: User,
+  organisation: Organisation
+): Promise<boolean> {
+  // find all userOnOrg where user is in the organisation OR user is an admin
+  const result = await prisma.userOnOrg.findMany({
+    where: {
+      OR: [
+        {
+          AND: [
+            {
+              user: user,
+            },
+            {
+              org: organisation,
+            },
+          ],
+        },
+        {
+          AND: [
+            {
+              user: user,
+            },
+            {
+              org: {
+                verified: true,
+              },
+            },
+          ],
+        },
+      ],
+    },
+  })
+  return result.length > 0
+}
+
+/*Returns number of minutes*/
+function convertDateToMinutes(date: Date): number {
+  return date.getTime() / (1000 * 60)
+}
+
+/**
+ * Checks that the user has selected at least the minimum number of slots
+ * and that the interval selected is in multiples of the slot size
+ *
+ * @param start
+ * @param end
+ * @returns true if  there is no violation
+ */
+export function checkStartEndTime(start: Date, end: Date): boolean {
+  return (
+    (convertDateToMinutes(end) - convertDateToMinutes(start)) /
+      DURATION_PER_SLOT >=
+      MIN_SLOTS_PER_BOOKING &&
+    (convertDateToMinutes(end) - convertDateToMinutes(start)) %
```

Comment:
Do we want to check that the start/end times are in intervals of for eg. 30 mins? or would the input be sanitised already. 

So for eg. if convertDateToMinutes(end) % DURATION_PER_SLOT != 0 we return false?

Reply:
ah i see, i only checked that the duration is in multiples of 30 mins. Good catch!